edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Edgelet/versioning/ModuleManagementHttpClientVersioned.cs

```diff
@@ -103,6 +108,13 @@ public virtual async Task<Stream> GetModuleLogs(string module, bool follow, Opti
 
         protected abstract void HandleException(Exception ex, string operation);
 
+        bool IsModuleManagementSocketStale(Exception ex)
+        {
+            return (ex is SocketException)
+                && (string.Compare(ex.GetType().ToString(), "System.Net.Internals.SocketExceptionFactory+ExtendedSocketException") == 0)
```

Comment:
This is not a good idea. We should just use the most derived public exception. 
We should not take a dependency on internal things like Internal Exception type or message.

Reply:
the most derived I can get would be `(ex is SocketException)` but that feels a little too generic.

Reply:
Please follow https://stackoverflow.com/questions/56909680/how-can-i-catch-an-extendedsocketexception to catch SocketException and check exception message.

Reply:
I think that's what I'm doing here. 
1. Check the exception is type of `SocketException`
2. Verify the messages
2.1 Make sure the type is of `System.Net.Internals.SocketExceptionFactory+ExtendedSocketException`
2.2 Observe the message that it is `Connection refused`

Reply:
As what Varun said you don't need to do 2.1.

Reply:
@varunpuranik please confirm if 2.2 is ok.

Reply:
I think the error message check should be fine. Given an example across 2.1 to 3.1, the error message didn't change but the statusCode changed.